app/services/content_item_publisher.rb

```diff
@@ -6,4 +6,48 @@ def publish(flow_presenters)
       Services.publishing_api.publish(content_item.content_id, 'minor')
     end
   end
+
+  def unpublish(content_id)
+    if content_id.present?
+      Services.publishing_api.unpublish(
+        content_id,
+        type: "gone",
+        unpublished_at: Time.now
+      )
+    else
+      raise "Content id has not been supplied"
+    end
+  end
+
+  def redirect_smart_answer(path, destination)
```

Comment:
This could be first iteration into decomposing this method. This way we understand what is happening and not how things are currently done when reading the method.

It would be good to document in the commit notes why we need to overwrite the segment mode in router-data, as it is not clear why this is happening for a reader with little context about what we are doing.

```ruby
def redirect_smart_answer(path, destination)
  raise "The destination or path isn't defined" unless path.present? && destination.present?

  add_redirect_to_publishing_api(path, destination)
  update_segment_mode_in_router_api(path, destination)
end

private

def add_redirect_to_publishing_api(path, destination)
  content_id = SecureRandom.uuid
  create_params = {
    base_path: path,
    document_type: :redirect,
    publishing_app: :smartanswers,
    schema_name: :redirect,
    redirects: [
      { path: path, type: :prefix, destination: destination }
    ]
  }

  response = Services.publishing_api.put_content(content_id, create_params)
  raise "This content item has not been created" unless response.code == 200
  Services.publishing_api.publish(content_id, :major)
end

def update_segment_mode_in_router_api(path, destination)
  Services.router_api.add_redirect_route(
    path,
    :prefix,
    destination,
    :permanent,
    segments_mode: :ignore
  )
end
```